### PR TITLE
Updated docs about calling out only velero-plugin-for-aws is compatible with vSphere plugin

### PR DESCRIPTION
**Pull request by lintongj**
Signed-off-by: Lintong Jiang <lintongj@vmware.com>

**What this PR does / why we need it**:

Update documents and call out only velero-plugin-for-aws is compatible with vSphere plugin

**Which issue(s) this PR fixes**:

Fixes https://github.com/vmware-tanzu/velero-plugin-for-vsphere/issues/300

**Special notes for your reviewer**: NONE

**Does this PR introduce a user-facing change?**: NONE

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/467/

**Review by deepakkinni — APPROVED** [review]
lgtm

File: docs/guest.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ The Data Manager Virtual Machine also needs to be installed to enable TKGS backu
 
 ### Install Object Storage Plugin
 
-Volume backups are stored in an object store bucket, e.g., S3. Currently these are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, please install and configure the object storage plugin.
-Here is an example, [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md).
+Volume backups are stored in an object store bucket, e.g., AWS S3. They are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, a Velero object storage plugin is required. Currently, only AWS plugin is supported and compatible with vSphere plugin.
```
Comment (xing-yang) [inline docs/guest.md:29]: Should we mention Minio somewhere?  Otherwise folks may think they can only use AWS S3.

Comment (lintongj) [reply docs/guest.md:29]: How about we can mention here that s3-compatible object stores can be supported via AWS plugin? Minio is just one of s3-compatible object store

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/468/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/469/

File: docs/vanilla.md [hunk 2]
```diff
@@ -36,9 +36,9 @@ of VDDK Programming Guide. **Note: please apply privileges at the vCenter Server
 
 ### Install Object Storage Plugin
 
-Volume backups are stored in an object store bucket, e.g., S3. Currently these are stored in the same bucket configured for
-the object storage plugin of Velero. Before installing the vSphere plugin, please install and configure the object storage plugin.
-Here is an example, [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md).
+Volume backups are stored in an object store bucket. They are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, a Velero object storage plugin is required.
+
+Currently, only AWS plugin is supported and compatible with vSphere plugin. Please refer to [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md) for more detail about using **AWS S3** as the object store for backups. S3-compatible object stores, e.g, **MinIO**, are also supported via AWS plugin. Please refer to [install with MinIO](https://velero.io/docs/v1.5/contributions/minio/).
```
Comment (xing-yang) [inline docs/vanilla.md:41]: nit: detail -> details

File: docs/guest.md [hunk 3]
```diff
@@ -26,8 +26,9 @@ The Data Manager Virtual Machine also needs to be installed to enable TKGS backu
 
 ### Install Object Storage Plugin
 
-Volume backups are stored in an object store bucket, e.g., S3. Currently these are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, please install and configure the object storage plugin.
-Here is an example, [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md).
+Volume backups are stored in an object store bucket. They are stored in the same bucket configured for the object storage plugin of Velero. Before installing the vSphere plugin, a Velero object storage plugin is required.
+
+Currently, only AWS plugin is supported and compatible with vSphere plugin. Please refer to [velero-plugin-for-aws](https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/master/README.md) for more detail about using **AWS S3** as the object store for backups. S3-compatible object stores, e.g, **MinIO**, are also supported via AWS plugin. Please refer to [install with MinIO](https://velero.io/docs/v1.5/contributions/minio/).
```
Comment (xing-yang) [inline docs/guest.md:31]: nit: detail -> details

File: docs/guest.md (hunk 1 above)
Comment (xing-yang) [reply docs/guest.md]: Yes, that sounds good.

**Comment by xing-yang** [comment]
Just some nits.

LGTM

**Review by xing-yang — APPROVED** [review]

**Pull request merged**